mirage/tls_mirage.mli

```diff
@@ -56,10 +56,15 @@ end
 
 (** X.509 handling given a key value store and a clock *)
 module X509 (KV : Mirage_kv.RO) (C : Mirage_clock.PCLOCK) : sig
-  (** [authenticator store clock typ] creates an [authenticator], either
-      using the given certificate authorities in the [store] or
-      null. *)
-  val authenticator : KV.t -> [< `Noop | `CAs ] -> X509.Authenticator.t Lwt.t
+  (** [authenticator ~hash_whitelist ~crl store typ] creates an [authenticator],
+      either using the given certificate authorities in the [store] as
+      value for key "ca_roots.crt", or null. If [hash_whitelist] is provided,
+      only these hashes are allowed for signatures of the certificate chain.
```

Comment:
```suggestion
      only these hash algorithms are allowed for signatures of the certificate chain.
```
just to remove any shred of doubt that it could be a list of hash values similar to the `Key_fingerprints` argument to X509_lwt.authenticator